src/lib.rs

```diff
@@ -80,6 +80,29 @@ pub mod pallet {
         BundlePublished { who: T::AccountId },
     }
 
+    #[pallet::genesis_config]
+    #[derive(frame_support::DefaultNoBound)]
+    pub struct GenesisConfig<T: Config> {
+        #[serde(skip)]
+        pub _phantom: core::marker::PhantomData<T>,
+
+        /// Use this option to override the default stdlib provided by the move-vm-backend.
+        pub change_default_stdlib_bundle_to: Option<Vec<u8>>,
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+        fn build(&self) {
+            let mut genesis_cfg = VmGenesisConfig::default();
+            if let Some(ref stdlib_bundle) = self.change_default_stdlib_bundle_to {
+                genesis_cfg.configure_stdlib(stdlib_bundle.clone());
+            }
+
+            let storage = Pallet::<T>::move_vm_storage();
+            genesis_cfg.apply(storage).unwrap();
```

Comment:
If we really want to terminate the whole runtime maybe at least we could use here `.expect()` with some message.

Reply:
True! I think the `assert` can also be used here

I think we can freely kill the whole runtime since this is happening at the genesis block creation, so it should be alright.

Reply:
Done